jdi-light/src/main/java/com/epam/jdi/light/settings/TestProperties.java

```diff
@@ -0,0 +1,281 @@
+    testsProperties
+            .put("requests.strategy", single(p -> LOGS.requestsStrategy = getLoggerStrategy(p)));
+    testsProperties.put("browser.kill", single(p -> COMMON.killBrowser = p));
+    testsProperties.put("element.search.strategy", single(TestProperties::setSearchStrategy));
+    testsProperties.put("browser.size", single(DRIVER.screenSize::read));
+    testsProperties
+            .put("page.load.strategy", single(p -> DRIVER.pageLoadStrategy = getPageLoadStrategy(p)));
+    testsProperties.put("page.check.after.open", single(p -> PAGE.checkPageOpen = parse(p)));
+    testsProperties.put("assert.type", single(SoftAssert::setAssertType));
+    testsProperties.put("click.type", single(p -> ELEMENT.clickType = getClickType(p)));
+    testsProperties.put("text.type", single(p -> ELEMENT.getTextType = getTextType(p)));
+    testsProperties.put("set.text.type", single(p -> ELEMENT.setTextType = getSetTextType(p)));
+///     RemoteWebDriver properties
+    testsProperties.put("remote.type", single(p -> DRIVER.remoteUrl = getRemoteUrl(p)));
+    testsProperties.put("driver.remote.url", single(p -> DRIVER.remoteUrl = p));
+    testsProperties.put("log.level",
+            combo(p -> LOGS.logLevel = parseLogLevel(p), () -> logger.setLogLevel(LOGS.logLevel)));
+    testsProperties.put("allure.steps", single(p -> LOGS.writeToAllure = parseBoolean(p)));
+    testsProperties.put("smart.locators", single(p -> ELEMENT.smartTemplate = p.split(";")[0]));
+    testsProperties
+            .put("smart.locators.toName", single(p -> ELEMENT.smartName = getSmartSearchFunc(p)));
+    testsProperties.put("smart.search", single(p -> ELEMENT.useSmartSearch = getSmartSearchUse(p)));
+    testsProperties.put("headless", single(p -> DRIVER.capabilities.common.put("headless", p)));
+  }
+
+  public static Pair<JAction1<String>, JAction> combo(JAction1<String> setPropertyAction,
+                                                      JAction additionalPropertyAction) {
+    return new Pair<>(setPropertyAction, additionalPropertyAction);
+  }
+
+  public static Pair<JAction1<String>, JAction> single(JAction1<String> setPropertyAction) {
+    return combo(setPropertyAction, _doNothing);
+  }
+
+  private static Strategies getStrategy(String prop) {
+    String strategy = prop.trim().toLowerCase();
+    switch (strategy) {
+      case "jdi":
+        return JDI;
+      case "selenium":
+        return SELENIUM;
+      default:
+        return JDI;
+    }
+  }
+
+  private static List<Strategy> getLoggerStrategy(String strategy) {
+    if (isBlank(strategy)) {
+      return new ArrayList<>();
+    }
+    List<Strategy> strategies = new ArrayList<>();
+    try {
+      String[] split = strategy.split(";");
+      strategies = LinqUtils.map(split, s -> parseStrategy(s.trim()));
+    } catch (Exception ignore) {
+    }
+    return strategies;
+  }
+
+  private static void setSearchStrategy(String p) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'setSearchStrategy(String)' has an NPath complexity of 580](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5524501)